dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramService.java

```diff
@@ -159,5 +159,27 @@
      */
     boolean hasOrgUnit( Program program, OrganisationUnit organisationUnit );
 
-    IdentifiableObjectAssociations getProgramOrganisationUnitsAssociations( Set<String> programUids );
+    /**
+     * Get all the organisation unit associated for a set of program uids. This
+     * method uses jdbc to directly fetch the associated org unit uids for every
+     * program uid. This method also filters the results to respect the current
+     * users organisation unit scope and sharing settings.
+     *
+     * @param programUids A set of program uids
+     * @return A object of {@link IdentifiableObjectAssociations} containing
+     *         association for each programUid
+     */
+    IdentifiableObjectAssociations getProgramOrganisationUnitsAssociationsForCurrentUser( Set<String> programUids );
+
+    /**
+     * Get all the organisation unit associated for a set of program uids. This
+     * method uses jdbc to directly fetch the associated org unit uids for every
+     * program uid. This method returns all the associations irrespective of the
+     * sharing settings or org unit scopes.
+     *
+     * @param programUids A set of program uids
+     * @return A object of {@link IdentifiableObjectAssociations} containing
+     *         association for each programUid
+     */
+    IdentifiableObjectAssociations getAllProgramOrganisationUnitsAssociations( Set<String> programUids );
```

Comment:
Perhaps remove the "All" part from the method name. Sort of cleaner without. After all it is not returning all, it is limited by the given program UIDs.

Reply:
Resolved. 
It was added to have a more visual difference from the other method which filters the "org units". This method fetches "All org unit associates for the specified programs".